lib/druid/data_source.rb

```diff
@@ -14,12 +15,21 @@ def initialize(name, uri)
       end
     end
 
-    def metadata
-      @metadata ||= metadata!
+    def metadata(opts = {})
+      @metadata ||= metadata!(opts)
```

Comment:
we either need to remove the cached `@metadata` here, or use the opts only with `metadata!` to prevent `opts` being ignored on subsequent calls